### DISCUSSION TITLE
New Ideas for webapp in v0.13

### DISCUSSION THREAD

**Discussion by alejandrogallo** [papis/papis#430] [Ideas]
I have a couple of ideas that I want to implement in the next coming months,
maybe someone else has requests/ideas.

### Add support for login and multiple accounts.

Like this you could setup in your research group a papis server
where you have students or coworkers access some libraries.
The loging should be configured depending on which rights
you give the users
- access which libraries?
- edit library information?
- [not implemented yet] add new documents?
- [not implemented yet] remove documents?
- [not implemented yet] update documents?

### Add support for citation-js

https://citation-js.github.io/

- I have already a POP implementation over at my fork.
    This is cool because we can then fetch metadata
    from crossref from only the browser, which then
    simplifies the backend.

- We can also fetch the citations from crossref
  from a given doi and then fetch the information of every citation.
  There is already some support of this in papis but it would
  be great also on demand from the webapp to fetch citation
  informations.
  
  
### Add an update document view

This is the old prolem from the TUI interface.
Namely, how do we ergonomically implement an interface that joins
metadata of two documents. Right now on the TUI interface
we are asked /à la/ `git add -p` what exactly we want from the other one.

My take for it is having the document view with two panes,
the left pane has an input array like right now in the picture
and the right pane has another such input array coming from
the importer source the info is coming from.

![doc-view](https://user-images.githubusercontent.com/12715358/204140979-7157b668-2bba-4c88-9799-9d450ed5e83b.jpg)

There are buttons to accept the changes brought from the right pane.

I would use rather this interface on my tablet while idle browsing
rather than the command line, curating my library.

**Comment by arunoruto** [comment]
I know the post was made a few years ago, but I wanted to ask if there is a solution for a shared library.
I am a PhD student, and I want a central location for the documents to be hosted (with the papis metadata) so everyone can access, edit, and add documents.

**Reply by alejandrogallo** [reply]
Hi, in principle this is planned but I have little time to work on that, but I'd really like something like that implemented, I might get to it soonish, but I can't really promise anything right now.
